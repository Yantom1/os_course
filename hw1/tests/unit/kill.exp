smash> smash error: kill: invalid arguments
smash> smash error: kill: job-id -1 does not exist
smash> smash error: kill: invalid arguments
smash> smash error: kill: job-id 1 does not exist
smash> smash error: kill: invalid arguments
smash> smash error: kill: invalid arguments
smash> smash error: kill: invalid arguments
smash> smash error: kill: job-id 1 does not exist
smash> smash> smash error: kill: invalid arguments
smash> smash error: kill: invalid arguments
smash> smash error: kill: invalid arguments
smash> smash error: kill: invalid arguments
smash> smash error: kill: invalid arguments
smash> signal number 9 was sent to pid \d+
smash> smash error: kill: job-id 1 does not exist
smash> smash> smash> smash error: kill failed: Invalid argument
smash> \[1\] sleep 10& : \d+ \d+ secs
smash> smash> \[1\] sleep 10& : \d+ \d+ secs
\[2\] sleep 5& : \d+ \d+ secs
smash> smash> \[1\] sleep 10& : \d+ \d+ secs
\[2\] sleep 5& : \d+ \d+ secs
\[3\] sleep 3& : \d+ \d+ secs
smash> smash error: kill: invalid arguments
smash> smash error: kill: invalid arguments
smash> smash> smash> \[1\] sleep 10 & : \d+ \d+ secs
smash> signal number 9 was sent to pid \d+
smash> smash> smash> smash> smash: got ctrl-Z
smash: process \d+ was stopped
smash> \[1\] sleep 5& : \d+ \d+ secs
\[2\]           sleep        10   & : \d+ \d+ secs
\[3\] sleep 15 : \d+ \d+ secs \(stopped\)
smash> signal number 9 was sent to pid \d+
smash> \[2\]           sleep        10   & : \d+ \d+ secs
\[3\] sleep 15 : \d+ \d+ secs \(stopped\)
smash> signal number 9 was sent to pid \d+
smash> \[3\] sleep 15 : \d+ \d+ secs \(stopped\)
smash> signal number 9 was sent to pid \d+
smash> smash> smash> signal number 19 was sent to pid \d+
smash> \[1\] sleep 10& : \d+ \d+ secs
smash> smash: got ctrl-C
smash: process \d+ was killed
smash> signal number 9 was sent to pid \d+
smash> smash: sending SIGKILL signal to 0 jobs:
