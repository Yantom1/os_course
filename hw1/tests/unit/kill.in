# invalid args
kill 9 1
kill -9 -1
kill -9 abc
kill -9 1
kill -abc 1
kill abc abc
kill -abc abc
kill -90 1
sleep 100&
kill -9 abc
kill -abc 1
kill abc abc
kill -abc abc
kill 9 1
kill -9 1
#test command format and validity of arguments - the next kill commands are invalid
kill -9 1
jobs
sleep 10&
kill -90 1
jobs
sleep 5&
jobs
sleep 3&
!time.sleep(1)
jobs 5
kill -not_a_number 1
kill -9 1 2 3
!time.sleep(11)
#jobs list should be empty now
jobs
sleep 10 &
!time.sleep(3)
jobs
kill -9 1
jobs junk_argument
sleep 5&
          sleep        10   &
sleep 15
CtrlZ
jobs
kill -9 1
jobs
kill -9 2
jobs
kill -9 3
jobs
sleep 10&
kill -19 1
!time.sleep(2)
jobs
!time.sleep(10)
sleep 10
!time.sleep(2)
CtrlC
kill -9 1
quit kill
