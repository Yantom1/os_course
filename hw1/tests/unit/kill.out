smash> smash error: kill: invalid arguments
smash> smash error: kill: job-id -1 does not exist
smash> smash error: kill: invalid arguments
smash> smash error: kill: job-id 1 does not exist
smash> smash error: kill: invalid arguments
smash> smash error: kill: invalid arguments
smash> smash error: kill: invalid arguments
smash> smash error: kill: job-id 1 does not exist
smash> smash> smash error: kill: invalid arguments
smash> smash error: kill: invalid arguments
smash> smash error: kill: invalid arguments
smash> smash error: kill: invalid arguments
smash> smash error: kill: invalid arguments
smash> signal number 9 was sent to pid 45259
smash> smash error: kill: job-id 1 does not exist
smash> smash> smash> smash error: kill failed: Invalid argument
smash> [1] sleep 10& : 45269 0 secs
smash> smash> [1] sleep 10& : 45269 0 secs
[2] sleep 5& : 45273 0 secs
smash> smash> [1] sleep 10& : 45269 2 secs
[2] sleep 5& : 45273 2 secs
[3] sleep 3& : 45276 1 secs
smash> smash error: kill: invalid arguments
smash> smash error: kill: invalid arguments
smash> smash> smash> [1] sleep 10 & : 45282 4 secs
smash> signal number 9 was sent to pid 45282
smash> smash> smash> smash> smash: got ctrl-Z
smash: process 45291 was stopped
smash> [1] sleep 5& : 45287 0 secs
[2]           sleep        10   & : 45289 0 secs
[3] sleep 15 : 45291 0 secs (stopped)
smash> signal number 9 was sent to pid 45287
smash> [2]           sleep        10   & : 45289 0 secs
[3] sleep 15 : 45291 0 secs (stopped)
smash> signal number 9 was sent to pid 45289
smash> [3] sleep 15 : 45291 1 secs (stopped)
smash> signal number 9 was sent to pid 45291
smash> smash> smash> signal number 19 was sent to pid 45300
smash> [1] sleep 10& : 45300 2 secs
smash> smash: got ctrl-C
smash: process 45304 was killed
smash> signal number 9 was sent to pid 45300
smash> smash: sending SIGKILL signal to 0 jobs:
